vowpalwabbit/cb_explore_adf.cc

```diff
@@ -358,48 +343,34 @@ void output_example(vw& all, cb_explore_adf& c, example& ec, v_array<example*>*
     all.print_text(all.raw_prediction, outputStringStream.str(), ec.tag);
   }
 
-  CB::print_update(all, is_test, ec, ec_seq, true);
+  CB::print_update(all, is_test, ec, &ec_seq, true);
 }
 
-void output_example_seq(vw& all, cb_explore_adf& data)
+void output_example_seq(vw& all, cb_explore_adf& data, multi_ex& ec_seq)
 {
-  if (data.ec_seq.size() > 0)
+  if (ec_seq.size() > 0)
   {
-    output_example(all, data, **(data.ec_seq.begin()), &(data.ec_seq));
+    output_example(all, data, ec_seq);
     if (all.raw_prediction > 0)
-      all.print_text(all.raw_prediction, "", data.ec_seq[0]->tag);
+      all.print_text(all.raw_prediction, "", ec_seq[0]->tag);
   }
 }
 
-
-void clear_seq_and_finish_examples(vw& all, cb_explore_adf& data)
-{
-  if (data.ec_seq.size() > 0)
-    for (example* ecc : data.ec_seq)
-      if (ecc->in_use)
-        VW::finish_example(all, ecc);
-  data.ec_seq.erase();
-}
-
-void finish_multiline_example(vw& all, cb_explore_adf& data, example& ec)
+void finish_multiline_example(vw& all, cb_explore_adf& data, multi_ex& ec_seq)
```

Comment:
In the long run, it would be good to unify finish_multiline_example and finish_example.   That seems beyond the scope at the moment.

Reply:
Need to understand this more.  Can work on this in future refactoring.